entry/src/entry.rs

```diff
@@ -485,55 +485,86 @@ pub fn start_verify_transactions(
+            // to have an overflow in the calculation (and using saturating_add would lead to
+            // an incorrect result in cases where an overflow would happen)
+            let num_vecs: usize = num_transactions / PACKETS_PER_BATCH
+                + if num_transactions % PACKETS_PER_BATCH != 0 {
+                    1
+                } else {
+                    0
+                };
+            let mut packets_vec = Vec::<Packets>::with_capacity(num_vecs);
+            let mut remaining_transactions = num_transactions;
+            for _n in 0..num_vecs {
+                let vec_size = cmp::min(PACKETS_PER_BATCH, remaining_transactions);
+                let mut packets = Packets::new_with_recycler(
+                    verify_recyclers.packet_recycler.clone(),
+                    vec_size,
+                    "entry-sig-verify",
+                );
+                // We use set_len here instead of resize(num_transactions, Packet::default()), to save
+                // memory bandwidth and avoid writing a large amount of data that will be overwritten
+                // soon afterwards. As well, Packet::default() actually leaves the packet data
+                // uninitialized anyway, so the initilization would simply write junk into
+                // the vector anyway.
+                unsafe {
+                    packets.packets.set_len(vec_size);
+                }
+                packets_vec.push(packets);
+                remaining_transactions -= vec_size;
             }
-
-            {
-                let entry_txs: Vec<&SanitizedTransaction> = entries
-                    .iter()
-                    .filter_map(|entry_type| match entry_type {
-                        EntryType::Tick(_) => None,
-                        EntryType::Transactions(transactions) => Some(transactions),
-                    })
-                    .flatten()
-                    .collect::<Vec<_>>();
-
-                let entry_tx_iter = entry_txs
+            assert!(remaining_transactions == 0);
+            let entry_txs: Vec<&SanitizedTransaction> = entries
+                .iter()
+                .filter_map(|entry_type| match entry_type {
+                    EntryType::Tick(_) => None,
+                    EntryType::Transactions(transactions) => Some(transactions),
+                })
+                .flatten()
+                .collect::<Vec<_>>();
+            let entry_txs = (0..num_vecs)
+                .map(|i: usize| {
+                    entry_txs[i * PACKETS_PER_BATCH
+                        ..i * PACKETS_PER_BATCH + packets_vec[i].packets.len()]
+                        .to_vec()
+                })
+                .collect::<Vec<_>>();
+            let res = packets_vec.par_iter_mut().zip(entry_txs).all(|pair| {
+                let entry_tx_iter = pair
+                    .1
                     .into_par_iter()
                     .map(|tx| tx.to_versioned_transaction());
```

Comment:
Could you use a flattened iterator over the packets and then zip with the `entry_tx_iter` you had before? It could be something like `Vec<&mut Packets>` I believe.